development/webpack/utils/cli.ts

```diff
@@ -0,0 +1,354 @@
+    .strict()
+    // use the scriptName in `--help` output
+    .scriptName(name)
+    // wrap output at a maximum of 120 characters or `process.stdout.columns`
+    .wrap(Math.min(120, process.stdout.columns))
+    // enable the `--config` command, which allows the user to specify a custom
+    // config file containing webpack options
+    .config()
+    .parserConfiguration({
+      'strip-aliased': true,
+      'strip-dashed': true,
+    })
+    // enable ENV parsing, which allows the user to specify webpack options via
+    // environment variables prefixed with `BUNDLE_`
+    // TODO: choose a better name than `BUNDLE` (it looks like `MM` is already being used in CI for ✨something✨)
+    .env('BUNDLE')
+    // TODO: enable completion once https://github.com/yargs/yargs/issues/2387 is fixed.
+    // enable the `completion` command, which outputs a bash completion script
+    // .completion(
+    //   'completion',
+    //   'Enable bash/zsh completions; concat the script generated by running this command to your .bashrc or .bash_profile',
+    // )
+    .example(
+      '$0 --env development --browser brave --browser chrome --zip',
+      'Builds the extension for development for Chrome & Brave; generate zip files for both',
+    )
+    // TODO: enable completion once https://github.com/yargs/yargs/issues/2387 is fixed.
+    // .example(
+    //   '$0 completion',
+    //   `Generates a bash completion script for the \`${name}\` command`,
+    // )
+    .updateStrings({
+      'Options:': toOrange('Options:'),
+      'Examples:': toOrange('Examples:'),
+    })
+    .options(options);
+  return cli;
+}
+
+type Options = ReturnType<typeof getOptions>;
+
+function getOptions(
+  isProduction: boolean,
+  buildTypes: string[],
+  allFeatures: string[],
+) {
+  const prodDefaultDesc = "If `env` is 'production', `true`, otherwise `false`";
+  return {
+    ...envOptions,
+    watch: {
+      alias: 'w',
+      array: false,
+      default: false,
+      description: 'Build then watch for files changes',
+      group: toOrange('Developer assistance:'),
+      type: 'boolean',
+    },
+    cache: {
+      alias: 'c',
+      array: false,
+      default: true,
+      description: 'Cache build for faster rebuilds',
+      group: toOrange('Developer assistance:'),
+      type: 'boolean',
+    },
+    progress: {
+      alias: 'p',
+      array: false,
+      default: true,
+      description: 'Show build progress',
+      group: toOrange('Developer assistance:'),
+      type: 'boolean',
+    },
+    devtool: {
+      alias: 'd',
+      array: false,
+      default: isProduction ? 'hidden-source-map' : 'source-map',
+      defaultDescription:
+        "If `env` is 'production', 'hidden-source-map', otherwise 'source-map'",
+      description: 'Sourcemap type to generate',
+      choices: ['none', 'source-map', 'hidden-source-map'] as const,
+      group: toOrange('Developer assistance:'),
+      type: 'string',
+    },
+    sentry: {
```

Comment:
There are couple of boolean CLI options here that don't have anything to do with the configuration of the _build_ system, but actually only serve as environment variables for the runtime. They are:

 * `--sentry` -> `process.env.ENABLE_SENTRY`
 * `--lavamoat` -> `process.env.ENABLE_LAVAMOAT`
 * `--lockdown` -> `process.env.ENABLE_LOCKDOWN`
 
 There are also CLI options that are used in HTML templates (which _are_ specifically defined by this build system) as well as environment variables for the runtime:
 
  * `--test` -> `process.env.IN_TEST` & HTML template's `it.isTest`
  * `--snow` `process.env.ENABLE_SNOW` & HTML template's `it.shouldIncludeSnow`
 
It might be a good idea to remove some of these in favor of environment variables only, just for consistency with other the way we use other environment variables. The `--test` and `--snow` options are tricky though, since they are so similar to the others, but must be read by the build system.

OR.... we could start moving away from using environment variables and making all options CLI options (in both gulp and webpack build systems).

Note: the webpack build system _does_ support using environment variables instead of options for build system configuration. You can do things like `BUNDLE_INTEST=1 yarn webpack` already.